### PR TITLE
Blokh/feat/assign case

### PR DESCRIPTION
**Pull request by Blokh**
### Description
The case functionality adds the the functionality of adding cases, assigning to a different user and implement read write logic

### Related issues
#306

### Breaking changes
this functionality provides actions logic and case's state for each assingment state and functionalities
### How these changes were tested
run kyc/kyb

### CODE REVIEW

File: apps/backoffice-v2/src/components/organisms/Subject/Subject.Actions.tsx [hunk 1]
```diff
@@ -194,6 +225,7 @@ export const Actions: FunctionComponent<IActionsProps> = ({ id, fullName, avatar
                   `btn-success btn justify-center before:mr-2 before:border-2 before:border-transparent before:content-[''] before:d-4 after:ml-2 after:border-2 after:border-transparent after:content-[''] after:d-4`,
                   {
                     loading: debouncedIsLoadingApproveEndUser,
+                    // 'opacity-50': isLoading || !canApprove
```
Comment (MatanYadaev) [inline apps/backoffice-v2/src/components/organisms/Subject/Subject.Actions.tsx:228]: Dead code. Uncomment or delete.

**Review by MatanYadaev — CHANGES_REQUESTED** [review]

File: apps/backoffice-v2/src/lib/react-query/mutations/useAssignWorkflowMutation/useAssignWorkflowMutation.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        },
+      }),
+    onMutate: ({
+      isAssignedToMe
+    }) => ({
+      resource: Resource.END_USER,
+      action:  isAssignedToMe ? Action.ASSIGNED_TO_ME : Action.ASSIGNED_TO_OTHER,
+    }),
+    onSuccess: () => {
+      queryClient.invalidateQueries();
```
[the remark this answers is not in the document]
Comment (Blokh) [reply apps/backoffice-v2/src/lib/react-query/mutations/useAssignWorkflowMutation/useAssignWorkflowMutation.tsx:20]: after discussion, decided to keep this

File: apps/backoffice-v2/src/api/types.ts [hunk 3]
```diff
@@ -1,6 +1,20 @@
 import { z } from 'zod';
 import { EndUserByIdSchema, EndUsersListSchema } from '../lib/zod/schemas/end-users';
+import { AuthenticatedUserSchema } from '../lib/zod/schemas/authenticated-user';
+import {UsersListSchema} from "../lib/zod/schemas/users";
 
 export type TEndUser = z.infer<typeof EndUserByIdSchema>;
 
 export type TEndUsers = z.infer<typeof EndUsersListSchema>;
+export type TUser = z.infer<typeof UsersListSchema>;
```
Comment (MatanYadaev) [inline apps/backoffice-v2/src/api/types.ts:9]:
```suggestion
export type TUsers = z.infer<typeof UsersListSchema>;
```

File: apps/backoffice-v2/src/lib/react-query/queries/useUsersQuery/useUsersQuery.tsx [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import { useQuery } from '@tanstack/react-query';
 import { users } from '../../users';
+import {TUser} from "../../../../api/types";
 
-export const useUsersQuery = () => {
+export const useUsersQuery = <TQueryFnData,>(
+  select: (data: {users: TUser}) => TQueryFnData,
```
Comment (MatanYadaev) [inline apps/backoffice-v2/src/lib/react-query/queries/useUsersQuery/useUsersQuery.tsx:6]: Do we support this in the back-end?

**Review by MatanYadaev — CHANGES_REQUESTED** [review]

**Review by MatanYadaev — APPROVED** [review]

**Pull request merged**